components/supervisor/frontend/src/index.ts

```diff
@@ -10,9 +10,38 @@
  */
 import { IDEMetricsServiceClient, MetricsName } from "./ide/ide-metrics-service-client";
 IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendClientTotal).catch(() => {})
-window.addEventListener('error', () => {
-    IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendErrorTotal).catch(() => {})
-})
+
+//#region supervisor frontend error capture
+declare global {
+    interface Window {
+        gitpodMetricsSupervisorFrontendErrorTotal(labels?: Map<string, string>): Promise<void>;
+    }
+}
+
+// Add this for other clients, i.e. openvscode-server
+window.gitpodMetricsSupervisorFrontendErrorTotal = async (labels?: Map<string, string>) => {
```

Comment:
we don't need it and should use browser API in VS Code Workbench directly, this file won't be necessary loaded at all